### PR TITLE
04 februarie tema5

### PR DESCRIPTION
**Pull request by marianpescaru**
Tests units for Functions homework

### CODE REVIEW

File: 16Ianuarie/src/main/java/com/sci/bv/Clock.java [hunk 1]
```diff
@@ -2,12 +2,14 @@
 
 import java.util.Scanner;
 
-public class Clock {
+public class Clock extends Exception{
```
Comment (georgeToaderSIIT) [inline 16Ianuarie/src/main/java/com/sci/bv/Clock.java:5]: why does clock extend exception?

File: 16Ianuarie/src/main/java/com/sci/bv/Fahrenheit.java [hunk 2]
```diff
@@ -2,12 +2,14 @@
 
 import java.util.Scanner;
 
-public class Fahrenheit {
+public class Fahrenheit extends Exception{
```
Comment (georgeToaderSIIT) [inline 16Ianuarie/src/main/java/com/sci/bv/Fahrenheit.java:5]: why does fahrenheit extend exception? is it an exception?

File: 16Ianuarie/src/main/java/com/sci/bv/Maximum.java [hunk 3]
```diff
@@ -0,0 +1,46 @@
+package com.sci.bv;
+
+import java.util.Scanner;
+
+public class Maximum extends Exception{
```
Comment (georgeToaderSIIT) [inline 16Ianuarie/src/main/java/com/sci/bv/Maximum.java:5]: again... why extends exception?

File: 16Ianuarie/src/test/java/com/sci/bv/FahrenheitTest.java [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+    @Test
+    public void fahrenheitCheckOK () {
+        int result = tempf.temperatureConversion(104);
+        Assertions.assertEquals(result,40);
+        // check if conversion temp F in temp C is correct
+        }
+
+    @Test
+    public void fahrenheitCheckNOK () {
+        Assertions.assertFalse(Fahrenheit.temperatureConversion(104)>40);
```
Comment (georgeToaderSIIT) [inline 16Ianuarie/src/test/java/com/sci/bv/FahrenheitTest.java:24]: I would have liked more values to be tested, other than 104... bigger values, negative values, 0

File: 16Ianuarie/src/test/java/com/sci/bv/MaximumTest.java [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+    }
+
+    @Test
+    void getMaxTrue(){
+        Assertions.assertTrue(Maximum.getMax(2,3)>2);
+    }
+
+    @Test
+    void getMaxFalse(){
+        Assertions.assertFalse(Maximum.getMax(4,5)<5);
```
Comment (georgeToaderSIIT) [inline 16Ianuarie/src/test/java/com/sci/bv/MaximumTest.java:26]: getMaxTrue and getMaxFalse aren't very good tests... you're not checking a single valid response, but that a response is a certain way.

To write more than one test here you could test equal values, test negative values, one negative and one positive...
